### PR TITLE
Chapter 8: Basic login

### PR DESCRIPTION
**Pull request by vietcrit96**
screenshot from 2019-01-03 08-34-49

### CODE REVIEW

**Comment by vietcrit96** [comment]
ready

File: app/assets/stylesheets/sessions.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Sessions controller here.
```
Comment (thanhhungchu95) [inline app/assets/stylesheets/sessions.scss]: remove this file

File: app/helpers/sessions_helper.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  def log_in user
+    session[:user_id] = user.id
+  end
+
+  def current_user
+    @current_user ||= User.find_by(id: session[:user_id]) if session[:user_id]
+  end
+
+  def logged_in?
+    !current_user.nil?
```
Comment (thanhhungchu95) [inline app/helpers/sessions_helper.rb]: use unless instead

File: app/helpers/sessions_helper.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  end
+
+  def log_out
+    session.delete(:user_id)
```
Comment (thanhhungchu95) [inline app/helpers/sessions_helper.rb]: remove the parentheses

File: app/models/user.rb [hunk 4]
```diff
@@ -9,4 +9,13 @@ class User < ApplicationRecord
...
   before_save{email.downcase!}
   has_secure_password
+
+  def self.digest string
+    cost = if ActiveModel::SecurePassword.min_cost
+             BCrypt::Engine::MIN_COST
+           else
+             BCrypt::Engine.cost
+           end
+    BCrypt::Password.create(string, cost: cost)
```
Comment (thanhhungchu95) [inline app/models/user.rb]: remove the parentheses

File: app/views/sessions/new.html.erb [hunk 5]
```diff
@@ -0,0 +1,19 @@
+<% provide :title, t("log_in")%>
+<h1><%= t "log_in" %></h1>
+
+<div class="row">
+  <div class="col-md-6 col-md-offset-3">
+    <%= form_with scope: :session, url: login_path, local: true do |f| %>
+
+      <%= f.label :email %>
+      <%= f.email_field :email, class: 'form-control' %>
```
Comment (thanhhungchu95) [inline app/views/sessions/new.html.erb]: use double-quote

File: app/views/sessions/new.html.erb [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+      <%= f.label :password, t(".pass")%>
+      <%= f.password_field :password, class: 'form-control' %>
```
Comment (thanhhungchu95) [inline app/views/sessions/new.html.erb]: same at here

File: config/locales/vi.yml [hunk 7]
```diff
@@ -37,6 +37,18 @@ vi:
...
+    new:
+      pass: "Mật khẩu"
+      new_user: "Tạo tài khoản mới? "
+  layouts:
+    header:
+      users: "Người dùng"
+      profile: "Thông tin"
+      setting: "Cài đặt"
+      log_out: "Đăng xuất"
+      language_html: Ngôn ngữ <b class="caret"></b>
```
Comment (thanhhungchu95) [inline config/locales/vi.yml]: use link_to do syntax

**Comment by vietcrit96** [comment]
ready

File: app/helpers/sessions_helper.rb [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+  def log_in user
+    session[:user_id] = user.id
+  end
+
+  def current_user
+    @current_user ||= User.find_by(id: session[:user_id]) if session[:user_id]
+  end
+
+  def logged_in?
+    unless current_user.nil?
```
Comment (thanhhungchu95) [inline app/helpers/sessions_helper.rb]: thôi, để là current_user.present? đi

**Review by thanhhungchu95 — COMMENTED** [review]
fix xong thì merged

**Pull request merged**